### ISSUE TITLE
UpdateSeriesAsync not working properly

### ISSUE THREAD

**Issue by soad-boon** [apexcharts/Blazor-ApexCharts#258]
Hello Joakim,

I am currently in the process of learning Blazor and found your Chart lib.
This seems to be the same/similiar issue as in #244 . The chart does not get updated correctly as it displays the previous values.

To reproduce the issue just create a Blazor-Server-project and in FetchData.razor add the following code.
Every time you click the Update Button the table refreshes with new values but the chart is stuck with the previous values. So the displayed chart data is 1 click behind.

Or am I doing simply something wrong?
Thanks for your help.
Chris

```
  <button @onclick="OnBtClick">Update</button>

  <table class="table">
    <thead>
    <tr>
      <th>Date</th>
      <th>Temp. (C)</th>
      <th>Temp. (F)</th>
      <th>Summary</th>
    </tr>
    </thead>
    <tbody>
    @foreach (var forecast in _forecasts)
    {
      <tr>
        <td>@forecast.Date.ToShortDateString()</td>
        <td>@forecast.TemperatureC</td>
        <td>@forecast.TemperatureF</td>
        <td>@forecast.Summary</td>
      </tr>
    }
    </tbody>
  </table>

  <ApexChart TItem="WeatherForecast"
             Title="Temp C"
             Height="300"
             @ref="_chart">

    <ApexPointSeries TItem="WeatherForecast"
                     Items="_forecasts"
                     Name="Temp C"
                     XValue="@(e => e.Date.ToString("g"))"
                     YAggregate="@(e => e.Sum(f => f.TemperatureC))"
                     SeriesType="SeriesType.Bar"
                     Color="#005ba3" />
  </ApexChart>
@code {
  private WeatherForecast[]? _forecasts;
  private ApexChart<WeatherForecast>? _chart;

  protected override async Task OnInitializedAsync()
  {
    await LoadAsync();
  }

  private async Task OnBtClick(MouseEventArgs args)
  {
    await LoadAsync();
  }

  private async Task LoadAsync()
  {
    _forecasts = await ForecastService.GetForecastAsync(DateTime.Now);
    if (_chart == null)
      return;

    await _chart.UpdateSeriesAsync();
  }
}
```

**Comment by nimrof** [comment]
Hi,
Got the same problem and i think it is a bug.
i got around it by not changing the 'Items' object once it is binded

So what i think should work for you is that you can just copy the result from GetForecastAsync into _forecasts instead of overwriting _forecasts.

**Comment by soad-boon** [comment]
Thanks for the workaround.
I can confirm that if you just add the items to the bound source then it works and the chart is updated correctly.

**Comment by DaleCam** [comment]
The solution here might also help: https://github.com/apexcharts/Blazor-ApexCharts/issues/269

**Comment by joadan** [comment]
Hello,

This should be fixed in v0.9.22-beta.
Please check sample https://apexcharts.github.io/Blazor-ApexCharts/methods/update-series
